### ISSUE TITLE
Tests fail with perl 5.23.4

### ISSUE THREAD

**Issue by eserte** [oliwer/mango#7]
t/bson.t fails with perl 5.23.4:
```
#   Failed test 'encode double NaN to double'
#   at t/bson.t line 190.
#          got: '#     expected: '
#   Failed test 'encode string NaN to double'
#   at t/bson.t line 191.
#          got: '#     expected: '# Looks like you failed 2 tests of 183.
t/bson.t ............... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/183 subtests 
```

Additionally, there are some warnings with this perl:
```
Hexadecimal number > 0xffffffff non-portable at t/bson.t line 207, <DATA> line 2125.
Hexadecimal number > 0xffffffff non-portable at t/bson.t line 228, <DATA> line 2125.
Hexadecimal number > 0xffffffff non-portable at t/bson.t line 230, <DATA> line 2125.
```

**Comment by eserte** [comment]
I have to revise my first analysis: I have a perl 5.23.4 (on FreeBSD 9.2) which generated a PASS report here. On three systems (Debian/wheezy, Debian/jessie, FreeBSD 10.1) a FAIL report was generated. There are also a couple of FAIL reports for earlier 5.23.x versions, but everything older generated PASS reports.

Maybe there will be more advice once http://analysis.cpantesters.org/solved?distv=Mango-1.21 has processed some reports...

**Comment by oliwer** [comment]
Thanks, I will look into this.

The warnings are normal because Mango only supports 64 bits OS. Otherwise
we would need to use Math::BigInt.

2015-11-04 20:35 GMT+01:00 Slaven Rezić <notifications@github.com>:

**Comment by oliwer** [comment]
After discussion on #p5p, it appears perl can store a NaN in multiple ways. In this case, the sign bit is different. Perl does not make a difference between +NaN and -NaN, nor should we.

**Comment by oliwer** [comment]
Solved in Mango v1.22.
